js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx

```diff
@@ -0,0 +1,67 @@
+import {Box, Icon} from '@dagster-io/ui-components';
+import {useMemo} from 'react';
+
+import {useStaticSetFilter} from './useStaticSetFilter';
+import {TruncatedTextWithFullTextOnHover} from '../../nav/getLeftNavItemsForOption';
+
+const emptyArray: any[] = [];
+
+export const useAssetOwnerFilter = ({
+  allAssetOwners,
+  owners,
+  setOwners,
+}: {
+  allAssetOwners: string[];
+  owners?: null | string[];
+  setOwners?: null | ((s: string[]) => void);
+}) => {
+  return useStaticSetFilter<string>({
+    name: 'Owner',
+    icon: 'account_circle',
+    allValues: useMemo(
+      () =>
+        allAssetOwners.map((value) => ({
+          value,
+          match: [value],
+        })),
+      [allAssetOwners],
+    ),
+    menuWidth: '300px',
+    renderLabel: ({value}) => (
+      <Box flex={{direction: 'row', gap: 4, alignItems: 'center'}}>
+        <Icon name="account_circle" />
+        <TruncatedTextWithFullTextOnHover tooltipText={value} text={value} />
+      </Box>
```

Comment:
You should use `UserDisplay` if it's a user imo

Reply:
Agreed. It's currently displaying strangely though -- I think we need to (1) unify OSS / Cloud user display styling and (2) Make a version that isn't a tag

<img width="416" alt="image" src="https://github.com/dagster-io/dagster/assets/29110579/85133453-d137-441e-8c18-a4c9c58ee71c">

What do you think about merging this version as is, and then switching to `UserDisplay` once the styling is updated?

Reply:
Hmm, try AuditLogUserDisplay

Reply:
I'm not opposed to making a new component for this similar to AuditLogUserDisplay (only in internal) for OSS

Reply:
The "claire" owner is the audit log user display, it also has the highlight on hover

If the owner doesn't exist as a cloud user, it displays using the oss user display. This is the "dataeng" owner in the picture

Reply:
I see, we should change the rendering logic to only use UserDisplay for users and then have something separate for non users

Reply:
yes

Reply:
I think if `isFilter` is true we should never use the tag version.

Reply:
eg here:
https://github.com/dagster-io/dagster/blob/f7d2611bdd46664496741b05bed3ce208d4f7b7e/js_modules/dagster-ui/packages/ui-core/src/runs/RunsFilterInput.tsx#L470

Reply:
sweet, thanks, updated